static/app/views/issueDetails/streamline/useEventDetailsReducer.tsx

```diff
@@ -0,0 +1,84 @@
+import {type Reducer, useCallback, useReducer} from 'react';
+
+import type {
+  SectionConfig,
+  SectionKey,
+} from 'sentry/views/issueDetails/streamline/context';
+
+export interface EventDetailsState {
+  searchQuery: string;
+  sectionData: {
+    [key in SectionKey]?: SectionConfig;
+  };
+}
+
+type OpenSectionAction = {
+  key: SectionKey;
+  type: 'OPEN_SECTION';
+};
+
+type UpdateSectionConfigAction = {
+  config: Partial<SectionConfig>;
+  key: SectionKey;
+  type: 'UPDATE_SECTION_CONFIG';
+};
+
+function updateSectionConfig(
+  state: EventDetailsState,
+  sectionKey: SectionKey,
+  updatedConfig: Partial<SectionConfig>
+): EventDetailsState {
+  const nextState = {
+    ...state,
+    sectionData: {
+      ...state.sectionData,
+    },
+  };
```

Comment:
its called nextState but doesn't seem to include the changes, i think it might make sense to create the new state here

Reply:
I just called it that because I mutate the object right after, so by the end `return nextState` makes sense.